context-free.md

```diff
@@ -1,22 +1,33 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+People who want to create primarily math/geometry-based images in a
+ (relatively) simple way. 
 
 ## What is easy to do in this language? Why is it easy?
-
+Creating images made up of a repitition of the same shape at a smaller
+size because the recursion available leads naturally to this.
 
 ## What is hard to do in this language? Why is it hard?
+Making images appear where you want them to is difficult because you
+either have to do a lot of math or a lot of experimentation. It is 
+also difficult to create a complex shape since the language only provides
+basic ones and as mentioned positioning is not intuitive. 
 
 
 ## How did you learn how to program in this language?
+By reading over the documentation on github and mostly by expirimenting
+and seeing what happened. 
 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
-
+It seems like it reads each instruction given separately and interprets it as
+an image, while storing the current "state" for the next instruction to be 
+interpreted in the context of.
 
```

Comment:
I think the idea of a "state" is a really important addition to the computational model. It really ties into the recursive nature of context-free. Also, I would add that after creating this "image" it attempts to determine its location on the canvas and then displays it.